modules/analytics-v2/src/backend/api.ts

```diff
@@ -0,0 +1,44 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import moment from 'moment'
+
+import { Analytics } from '..'
+
+import { AnalyticsDatabase } from './db'
+
+export default (bp: typeof sdk, db: AnalyticsDatabase) => {
+  const router = bp.http.createRouterForBot('analytics-v2')
+
+  router.get('/channel/:channel', async (req, res) => {
+    const { botId, channel } = req.params
+    const { start, end } = req.query
+
+    const startDate = unixToDate(start)
+    const endDate = unixToDate(end)
+
+    try {
+      if (!channel || channel === 'all') {
+        const analytics = await db.getBetweenDates(botId, startDate, endDate, undefined)
+        res.send(analytics.map(toDto))
```

Comment:
I really don't want to find issues where there aren't, this is more a general comment then anything specific to this code.  Just so you know there's a security issue of returning an array to the client (learn more about this [here](https://haacked.com/archive/2009/06/25/json-hijacking.aspx/)) therefore you should always try to nest your payload in an object.

Reply:
The method doesn't fit the first "requirement" to make the vulnerability worthwile ("returns sensitive data") so it's not really an issue there, but still a good point to keep in mind